app/src/main/kotlin/com/softteco/template/ui/components/snackbar/SnackbarController.kt

```diff
@@ -0,0 +1,37 @@
+package com.softteco.template.ui.components.snackbar
+
+import androidx.annotation.StringRes
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SnackbarController @Inject constructor() {
+
+    private val _snackbars = MutableStateFlow<Set<SnackbarState>>(emptySet())
+    val snackbars get() = _snackbars.asStateFlow()
+
+    private val _onDismissEvents = MutableSharedFlow<SnackbarState>()
```

Comment:
How do we use _onDismissEvents?

Reply:
Not used in any way in the current cases. For example, you can implement navigation to another screen only after Snackbar was dismissed (after error or hint)